pkg/scheduler/framework/plugins/helper/spread_test.go

```diff
@@ -99,3 +101,180 @@ func TestGetPodServices(t *testing.T) {
 		})
 	}
 }
+
+func TestDefaultSelector(t *testing.T) {
+	const (
+		namespace                 = "test"
+		podName                   = "test-pod"
+		serviceName               = "test-service"
+		replicaSetName            = "test-replicaset"
+		replicationControllerName = "test-replicationcontroller"
+		statefulSetName           = "test-statefulset"
+
+		podLabelKey = "podLabelKey"
+		podLabelVal = "podLabelVal"
+
+		replicaSetLabelKey = "replicaSetLabelKey"
+		replicaSetLabelVal = "replicaSetLabelVal"
+
+		replicationLabelKey = "replicationLabelKey"
+		replicationLabelVal = "replicationLabelVal"
+
+		statefulSetLabelKey = "statefulSetLabelKey"
+		statefulSetLabelVal = "statefulSetLabelVal"
+	)
+
+	fakeInformerFactory := informers.NewSharedInformerFactory(&fake.Clientset{}, 0*time.Second)
+
+	// Create fake service
+	addFakeService := func() error {
+		// Create fake service
+		service := &v1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      serviceName,
+				Namespace: namespace,
+			},
+			Spec: v1.ServiceSpec{
+				Selector: map[string]string{
+					podLabelKey: podLabelVal,
+				},
+			},
+		}
+		return fakeInformerFactory.Core().V1().Services().Informer().GetStore().Add(service)
+	}
+
+	// Create fake ReplicaSet
+	addFakeReplicaSet := func() error {
+		replicaSet := &appsv1.ReplicaSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      replicaSetName,
+				Namespace: namespace,
+			},
+			Spec: appsv1.ReplicaSetSpec{
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						replicaSetLabelKey: replicaSetLabelVal,
+					},
+				},
+			},
+		}
+		return fakeInformerFactory.Apps().V1().ReplicaSets().Informer().GetStore().Add(replicaSet)
+	}
+
+	// Create fake ReplicationController
+	addFakeReplicationController := func() error {
+		replicationController := &v1.ReplicationController{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      replicationControllerName,
+				Namespace: namespace,
+			},
+			Spec: v1.ReplicationControllerSpec{
+				Selector: map[string]string{
+					replicationLabelKey: replicationLabelVal,
+				},
+			},
+		}
+		return fakeInformerFactory.Core().V1().ReplicationControllers().Informer().GetStore().Add(replicationController)
+	}
+
+	// Create fake StatefulSet
+	addFakeStatefulSet := func() error {
+		statefulSet := &appsv1.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      statefulSetName,
+				Namespace: namespace,
+			},
+			Spec: appsv1.StatefulSetSpec{
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						statefulSetLabelKey: statefulSetLabelVal,
+					},
+				},
+			},
+		}
+		return fakeInformerFactory.Apps().V1().StatefulSets().Informer().GetStore().Add(statefulSet)
+	}
+
+	tests := []struct {
+		name            string
+		pod             *v1.Pod
+		expect          labels.Set
+		addFakeService  func() error
```

Comment:
Why we need this? Doesn't this can also be part of `addFakeResource`.

Reply:
I changed addFakeResourceList to slice, and passed the func of service and other resources respectively.
```go
tests := []struct {
		name                string
		pod                 *v1.Pod
		expect              labels.Set
		addFakeResourceList []func() error
	}
```